network/retrieval/retrieve_test.go

```diff
@@ -172,7 +172,7 @@ func TestUnsolicitedChunkDelivery(t *testing.T) {
 		})
 
 	// expect peer disconnection
-	err = tester.TestDisconnected(&p2ptest.Disconnect{Peer: node.ID(), Error: errors.New("subprotocol error")})
+	err = tester.TestDisconnected(&p2ptest.Disconnect{Peer: node.ID(), Error: errors.New("Message handler error: (msg code 0): unsolicited chunk delivery from peer: cannot find ruid")})
```

Comment:
This is a difference in disconnect errors now that I mentioned in 3. in the description.

Reply:
@zelig It appears you were right. The disconnect will eventually happen with `subprotocol error` anyway. But, if we return the specific error from the run loop, instead of doing a `peer.Stop()`, the actual error  will be broadcasted via peer broadcast. I am still not sure if it is useful to us, but it's still probably more clean.